client/extensions/woocommerce/state/sites/settings/stripe-connect-account/handlers.js

```diff
@@ -28,22 +36,51 @@ export function handleAccountCreate( { dispatch }, action ) {
 }
 
 export function handleAccountCreateSuccess( store, action, { data } ) {
-	const { email, siteId } = action;
+	const { siteId } = action;
 	const { account_id } = data;
 
 	store.dispatch( {
 		type: WOOCOMMERCE_SETTINGS_STRIPE_CONNECT_ACCOUNT_CREATE_COMPLETE,
 		connectedUserID: account_id,
-		email,
 		siteId,
 	} );
 }
 
 export function handleAccountCreateFailure( { dispatch }, action, error ) {
-	const { email, siteId } = action;
+	const { siteId } = action;
 	dispatch( {
 		type: WOOCOMMERCE_SETTINGS_STRIPE_CONNECT_ACCOUNT_CREATE_COMPLETE,
+		error,
+		siteId,
+	} );
+}
+
+export function handleAccountFetch( { dispatch }, action ) {
+	const { siteId } = action;
+	dispatch( request( siteId, action, '/wc/v1' ).get( 'connect/stripe/account/', {} ) );
+}
+
+export function handleAccountFetchSuccess( store, action, { data } ) {
+	const { siteId } = action;
+	const { account_id, display_name, email, business_logo, legal_entity, payouts_enabled } = data;
+
+	store.dispatch( {
+		type: WOOCOMMERCE_SETTINGS_STRIPE_CONNECT_ACCOUNT_DETAILS_UPDATE,
+		connectedUserID: account_id,
+		displayName: display_name,
 		email,
+		firstName: legal_entity.first_name,
```

Comment:
can we always trust that the API will return a `legal_entity` object or should we add a check here and for `last_name` below?